test/constraintViolations/src/test/java/com/mvcspec/ozark/test/constraintViolations/ConstraintViolationsTest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2017 Ivar Grimstad (ivar.grimstad@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mvcspec.ozark.test.constraintViolations;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import javax.validation.constraints.Pattern;
+import javax.ws.rs.FormParam;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mvcspec.ozark.binding.validate.ConstraintViolationMetadata;
+import org.mvcspec.ozark.binding.validate.ConstraintViolations;
+import org.mvcspec.ozark.test.constraintViolations.BaseUser;
+import org.mvcspec.ozark.test.constraintViolations.MvcUser;
+import org.mvcspec.ozark.test.constraintViolations.NotAJavaBean;
+import org.mvcspec.ozark.test.constraintViolations.TestAnnotation;
+
+/**
+ * Unit tests for annotation resolution in {@link ConstraintViolations}.
+ * <p>
+ * These tests are based on a scenario where we have a number of existing java
+ * bean classes that have bean validation constraints already applied to both
+ * fields and getter methods. We extend those beans so that we can annotate the
+ * field getter methods with {@link TestAnnotation} (which simulates the {@link
+ * javax.mvc.binding.MvcBinding} annotation), and annotate the setter methods
+ * with {@link javax.ws.rs.FormParam}, as would be done on a JAX-RS/MVC {@link
+ * javax.ws.rs.BeanParam}.
+ */
+public class ConstraintViolationsTest {
```

Comment:
The test looks great! Thanks a lot for adding this. Having more test coverage is awesome!

But actually the `tests` directory is just for integration tests which contain real web application which can be deployed to server to run asserts from "outside". I know that `tests` directory name isn't optimal, but it was a very early decision to call it this way and we didn't rename it yet.

So feel free to move the tests to `core/src/test/java`..